samples/image-metadata/src/components/FocalPointDialog/FocalPointDialog.js

```diff
@@ -0,0 +1,144 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Subheading, Modal } from '@contentful/forma-36-react-components';
+
+import { FocalPoint } from '../FocalPoint';
+import { ImagePreviewWithFocalPoint } from '../ImagePreviewWithFocalPoint';
+
+import { MAX_PREVIEW_WRAPPER_SIZE, styles } from './styles';
+
+export class FocalPointDialog extends Component {
+  static propTypes = {
+    file: PropTypes.object.isRequired,
+    focalPoint: PropTypes.shape({
+      x: PropTypes.number.isRequired,
+      y: PropTypes.number.isRequired
+    }),
+    onClose: PropTypes.func.isRequired,
+    onSave: PropTypes.func.isRequired,
+    sdk: PropTypes.object.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+    this.imgRef = React.createRef();
+  }
+
+  state = {
+    focalPoint: this.props.focalPoint || { x: 0, y: 0 },
+    imgElementRect: null
+  };
+
+  getAdjustedFocalPointForUI = () => {
+    const { file } = this.props;
+    const { focalPoint, imgElementRect } = this.state;
+
+    const { width, height } = file.details.image;
+    const widthRatio = width / imgElementRect.width;
+    const heightRatio = height / imgElementRect.height;
+
+    const marginLeft = Math.max((MAX_PREVIEW_WRAPPER_SIZE - imgElementRect.width) / 2, 0);
+    const marginTop = Math.max((MAX_PREVIEW_WRAPPER_SIZE - imgElementRect.height) / 2, 0);
+
+    return {
+      x: Math.round(focalPoint.x / widthRatio + marginLeft),
+      y: Math.round(focalPoint.y / heightRatio + marginTop)
+    };
+  };
+
+  onImageClick = e => {
+    const { file } = this.props;
+
+    const rect = e.target.getBoundingClientRect();
+    const x = e.clientX - rect.left;
+    const y = e.clientY - rect.top;
+
+    const { width, height } = file.details.image;
+
+    const widthRatio = width / rect.width;
+    const heightRatio = height / rect.height;
+
+    const actualX = Math.round(x * widthRatio);
+    const actualY = Math.round(y * heightRatio);
+
+    this.setState({
+      focalPoint: { x: actualX, y: actualY }
+    });
+  };
+
+  onImageLoad = e => {
+    this.setState({
+      imgElementRect: e.target.getBoundingClientRect()
```

Comment:
Does this need updating on window resize?

Reply:
It did :) Done.